src/test/resources/com/jetbrains/snakecharm/cucumber/resolve/localrules_and_ruleorder.feature

```diff
@@ -0,0 +1,51 @@
+Feature: Resolve for rules in localrules and ruleorder
+
+  Scenario Outline: Resolve in localrules section
+    Given a snakemake project
+    Given I open a file "foo.smk" with text
+    """
+    rule aaaa:
+      input: "input.txt"
+
+    rule bbbb:
+      output: touch("output.txt")
+
+    rule cccc:
+      output: touch("_output.txt")
+
+    localrules: <text>
+    """
+    When I put the caret after <ptn>
+    Then reference should resolve to "<symbol_name>" in "<file>"
+
+    Examples:
+      | ptn             | text      | symbol_name | file         |
+      | localrules: aaa | aaaa      | aaaa        | foo.smk      |
+      | localrules: bbb | bbbb      | bbbb        | foo.smk      |
+      | localrules: ccc | cccc      | cccc        | foo.smk      |
```

Comment:
Please check case `localrules: aaa, b<caret>bb`. Also check that resolve works when rule declaration is in the beginning of file before other rules.